src/Serilog.Sinks.AzureTableStorage/Sinks/AzureTableStorage/AzureBatchingTableStorageSink.cs

```diff
@@ -51,7 +53,7 @@ public AzureBatchingTableStorageSink(CloudStorageAccount storageAccount, IFormat
             if (string.IsNullOrEmpty(storageTableName)) storageTableName = typeof(LogEventEntity).Name;
 
             _table = tableClient.GetTableReference(storageTableName);
-            _table.CreateIfNotExists();
+            _table.CreateIfNotExistsAsync().Wait(_waitTimeoutMilliseconds);
```

Comment:
If this is ever called from code with a synchronisation context it will cause a deadlock. Is that possible or will this always be called in the background by Serilog?